src/assisted_test_infra/test_infra/utils/utils.py

```diff
@@ -560,6 +561,37 @@ def get_assisted_controller_status(kubeconfig):
     return response.stdout
 
 
+def retry_success(repeat, sleep_between=30):
+    def retry_func(func):
+        def inner(*args, **kwargs):
+            start = 1
+            while start <= repeat:
+                log.info(f"retry_success decorator call func {func.__name__} {start} time")
+                res = func(*args, **kwargs)
+                if res:
+                    return res
+                else:
+                    start += 1
+                    time.sleep(sleep_between)
+            return ""
+
+        return inner
+
+    return retry_func
+
+
+@retry_success(5)
+def get_assisted_controller_log(kubeconfig, log_resource):
+    log.info("Getting controller logs")
```

Comment:
maybe add more context to the log line with the `kubeconfig` and `log_resources` ?